### PR TITLE
febio: 3.6 -> 4.7; febio-studio: 1.6.1 -> 2.7

### PR DESCRIPTION
**Pull request by emilazy**
## Description of changes

Phew, this was a lot of work! Another FFmpeg 4 dependency upgraded, and some Darwin support to sweeten the deal.

@Scriptkiddi I hope you still use this! I have absolutely no idea how to operate the program, but I’ve checked that it starts and can call out to FEBio and so on. It would be great if you could test it more thoroughly, especially on Linux.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fe/febio-studio/package.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  ninja,
+  zlib,
+  libGLU,
+  qt6Packages,
+  febio,
+  glew,
+  sshSupport ? true,
+  openssl,
+  libssh,
+  tetgenSupport ? true,
```
Comment (philiptaron) [inline pkgs/by-name/fe/febio-studio/package.nix:16]: confirmed that this builds with `tetgenSupport` set to false

File: pkgs/by-name/fe/febio-studio/package.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  tetgen,
+  ffmpegSupport ? true,
+  ffmpeg_7,
+  dicomSupport ? false,
+  dcmtk,
+  withModelRepo ? true,
```
Comment (philiptaron) [inline pkgs/by-name/fe/febio-studio/package.nix:22]: confirmed that this builds with `withModelRepo` set to false

**Review by philiptaron — APPROVED** [review]
Gave a thorough build check.

**Comment by philiptaron** [comment]
I intend to merge on Monday, in order to give @Scriptkiddi time to look. But this seems 👍🏻

File: pkgs/by-name/fe/febio/package.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  stdenv,
+  overrideSDK,
+  fetchFromGitHub,
+  fetchpatch2,
+  substituteAll,
+  cmake,
+  ninja,
+  zlib,
+  darwin,
+  mklSupport ? true,
```
[the remark this answers is not in the document]
Comment (emilazy) [reply pkgs/by-name/fe/febio/package.nix:12]: Yeah, I get the impression that you’re really just meant to use it:

> FEBio requires the Intel Math Kernel Library (MKL) in order to utilize the Pardiso linear solver and some of the iterative linear solvers. This library can be downloaded as part of the Intel oneAPI Base Toolkit from Intel's website: https://software.intel.com/content/www/us/en/develop/tools/oneapi/base-toolkit.html. In the absence of MKL, FEBio will default to using the Skyline linear solver. However, the Pardiso solver is significantly faster and more memory-efficient than the Skyline solver, and it is strongly recommended that the Pardiso solver be used.

File: pkgs/by-name/fe/febio/package.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  src = fetchFromGitHub {
+    owner = "febiosoftware";
+    repo = "FEBio";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-RRdIOyXg4jYW76ABfJdMfVtCYMLYFdvyOI98nHXCof8=";
+  };
+
+  patches = [
+    # Fix library searching and installation
+    (substituteAll {
```
[the remark this answers is not in the document]
Comment (emilazy) [reply pkgs/by-name/fe/febio/package.nix:33]: Thanks for the pointers! https://github.com/NixOS/nixpkgs/pull/301350 looks interesting. This is just inherited from the previous version of the derivation, though.

[the remark this answers is not in the document]
Comment (philiptaron) [reply pkgs/by-name/fe/febio/package.nix:33]: Yeah, hence "unrelated" -- no expectation for you to take any action.

**Comment by Scriptkiddi** [comment]
thanks for taking care of this, im not using febio really active anymore but I know its being used for research by 8 people

**Pull request merged**